### PR TITLE
Feature/beer api#65

### LINKED ISSUE #65: Functionality for beer api

### PR DESCRIPTION
**Pull request by edman193**
closes #65 

Synchronize beer catalog with api catalog

### CODE REVIEW

File: app/jobs/sync_beer_catalog_job.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    @options = stored_options
+    @current_state = @options['beers']['state']
+    @current_available = @options['beers']['available_id']
+    @current_available_idx = 0
+    @current_page = @options['beers']['page']
+    @total_pages = nil
+    @beers = []
+
+    @available = brewery_db.menu.beer_availability || []
+    @available_ids =  @available.take(@available.size).select { |a| a['id'] >= @current_available }.map { |a| a['id'] }
```
Comment (jclopezdev) [inline app/jobs/sync_beer_catalog_job.rb]: I think @jnegrete is going to get mad over this line

File: app/jobs/sync_beer_catalog_job.rb [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          photo_remote_url: (beer['labels'] && beer['labels']['large'] || nil)
+        } if !local_beers.include?(beer['id'])
+      end
+      return true
+    end
+    return false
+  end
+
+  def save_beers
+    binding.pry
```
Comment (jclopezdev) [inline app/jobs/sync_beer_catalog_job.rb]: 🤔

File: app/services/brewerydb_service.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'httparty'
+require 'httplog'
+class BrewerydbService
+  include HTTParty
+  base_uri 'api.brewerydb.com'
+
+  def initialize(available_id, page)
+    @options = { query: { availableId: available_id, p: page, key: ENV['BREWERYDB_APIKEY']} }
```
Comment (jclopezdev) [inline app/services/brewerydb_service.rb]: Don't forget tu use ENV.fetch() instead of only ENV[]

File: app/services/brewerydb_service.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+require 'httparty'
+require 'httplog'
+class BrewerydbService
+  include HTTParty
+  base_uri 'api.brewerydb.com'
```
Comment (jclopezdev) [inline app/services/brewerydb_service.rb]: @edman193 
What do you think if we add a environment variable for the base_uri of the brewery api?

File: app/views/admin/beers/_collection.html.erb [hunk 5]
```diff
@@ -0,0 +1,96 @@
+<%#
```
Comment (jclopezdev) [inline app/views/admin/beers/_collection.html.erb]: What about haml? 🤔

File: app/views/admin/beers/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,67 @@
+<%#
```
Comment (jclopezdev) [inline app/views/admin/beers/index.html.erb]: What about haml?

File: db/migrate/20180327185717_add_photo_remote_url_to_beers.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+class AddPhotoRemoteUrlToBeers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :beers, :photo_remote_url, :string
```
Comment (jclopezdev) [inline db/migrate/20180327185717_add_photo_remote_url_to_beers.rb]: Why is it necessary to add another field for the image?

File: test/jobs/sync_beer_catalog_job_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (jclopezdev) [inline test/jobs/sync_beer_catalog_job_test.rb]: Could you delete this file?

Comment (edman193) [reply test/jobs/sync_beer_catalog_job_test.rb]: mmm, We have to add some tests, but for now ok will delete it

File: db/migrate/20180327185717_add_photo_remote_url_to_beers.rb (hunk 7 above)
Comment (jclopezdev) [reply db/migrate/20180327185717_add_photo_remote_url_to_beers.rb]: Ok 👍

**Comment by jclopezdev** [comment]
image

**Pull request merged**